apps/crn-server/test/data-providers/contentful/working-groups.data-provider.test.ts

```diff
@@ -0,0 +1,651 @@
+                    alumniSinceDate: null,
+                  },
+                },
+              ],
+            },
+          },
+        };
+
+        contentfulGraphqlClientMock.request.mockResolvedValueOnce(
+          contentfulGraphQLResponse,
+        );
+
+        const result = await workingGroupDataProvider.fetchById(id);
+
+        expect(result?.leaders).toHaveLength(1);
+        expect(result?.leaders[0]?.user).toEqual({
+          alumniSinceDate: null,
+          avatarUrl: undefined,
+          displayName: 'John Doe',
+          email: 'johndoe@gmail.com',
+          firstName: 'John',
+          id: 'leader-1',
+          lastName: 'Doe',
+        });
+        expect(result?.members).toHaveLength(1);
+        expect(result?.members[0]?.user).toEqual({
+          alumniSinceDate: null,
+          avatarUrl: 'https://images.ctfassets.net/a01453c.png',
+          displayName: 'Jane Doe',
+          email: 'jane@gmail.com',
+          firstName: 'Jane',
+          id: 'member-1',
+          lastName: 'Doe',
+        });
+      });
+
+      test('should return empty leaders and members if they do not exist', async () => {
+        const id = 'some-id';
+
+        const contentfulGraphQLResponse = getContentfulGraphqlWorkingGroup();
+        contentfulGraphQLResponse.membersCollection = null;
+
+        contentfulGraphqlClientMock.request.mockResolvedValueOnce({
+          workingGroups: contentfulGraphQLResponse,
+        });
+
+        const result = await workingGroupDataProvider.fetchById(id);
+
+        expect(result?.members).toStrictEqual([]);
+        expect(result?.leaders).toStrictEqual([]);
+      });
+
+      test('should filter null membersCollection items', async () => {
+        const id = 'some-id';
+        const contentfulGraphQLResponse = {
+          workingGroups: {
+            ...getContentfulGraphqlWorkingGroup(),
+            membersCollection: {
+              items: [
+                null,
+                {
+                  __typename: 'WorkingGroupMembers',
+                  inactiveSinceDate: '2023-05-25T00:00:00.000-03:00',
+                  user: {
+                    sys: {
+                      id: 'member-1',
+                    },
+                    avatar: {
+                      url: 'https://images.ctfassets.net/a01453c.png',
+                    },
+                    email: 'jane@gmail.com',
+                    firstName: 'Jane',
+                    lastName: 'Doe',
+                    alumniSinceDate: null,
+                  },
+                },
+              ],
+            },
+          },
+        };
+
+        contentfulGraphqlClientMock.request.mockResolvedValueOnce(
+          contentfulGraphQLResponse,
+        );
+
+        const result = await workingGroupDataProvider.fetchById(id);
+
+        expect(result?.members).toHaveLength(1);
+        expect(result?.leaders).toHaveLength(0);
+      });
+    });
+  });
+
+  describe('Update method', () => {
```

Comment:
for how complex the method implementation is i think the test coverage is quite low, don't you think? there are a lot of edge cases here like - what happens when there is an error half way through removing old deliverables etc. Not sure if you were planning to add these or are we just going with a simple approach for now until its a problem?

Reply:
I've added some try catch blocks. I don't think it's a major issue not deleting the old entries, so I've just added a log when this happens. But when it fails to create a deliverable it would throw [adding try-catch blocks in update method](https://github.com/yldio/asap-hub/pull/3110/commits/aca564ceb11a3b54347bfeb2cc708b8ca974fdc2)